pkg/pipeline/utils/exit.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+var (
+	registeredChannels []chan bool
+	chanMutex          sync.Mutex
+)
+
+func RegisterExitChannel(ch chan bool) {
+	chanMutex.Lock()
+	registeredChannels = append(registeredChannels, ch)
+	chanMutex.Unlock()
```

Comment:
can you change to `defer chanMutex.Unlock()` instead of writing in the end ?

Reply:
done